hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/ProxyServiceImpl.java

```diff
@@ -102,6 +99,7 @@ public void handleException(Throwable throwable) {
 
     public ProxyServiceImpl(NodeEngineImpl nodeEngine) {
         this.nodeEngine = nodeEngine;
+        this.destroyTimeout = parseInt(System.getProperty("hazelcast.proxy.destroy.timeout.millis", "30"));
```

Comment:
Do we specify timeout in seconds or milliseconds? The name suggests milliseconds, but we interpret it in seconds?